File: ClientApp/src/app/modules/Product.ts
import { Attributes } from "./Attributes";

export class Product {

  id: number | null;
  name: string;
  price: number;
  attr: Attributes;


  constructor() {
    this.id = null;
    this.price = 0;
    this.name = "";
    this.attr = new Attributes;

  }
}

File: Backend/src/main/java/com/ShopWebApplicationBackend/Backend/entities/Product.java
package com.ShopWebApplicationBackend.Backend.entities;
import javax.persistence.*;
import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Table;


@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue
    private Long id;

    private String name;

    private Long price;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL, mappedBy = "product")
    @JoinColumn(name = "attributes_id", nullable = false)
    private Attributes attr;


}
